services/catalog/src/simcore_service_catalog/api/rest/_services.py

```diff
@@ -105,56 +108,67 @@ async def list_services(
         )
 
     # now get the executable or writable services
-    services_in_db = {
-        (s.key, s.version): s
-        for s in await services_repo.list_services(
-            gids=[group.gid for group in user_groups],
-            execute_access=True,
-            write_access=True,
-            combine_access_with_and=False,
-            product_name=x_simcore_products_name,
-        )
-    }
+    with _tracer.start_as_current_span("services_repo.list_services") as span:
+        span.set_attribute("osparc-gids", f"{[group.gid for group in user_groups]}")
+        span.set_attribute("osparc-productname", f"{x_simcore_products_name}")
+        services_in_db = {
+            (s.key, s.version): s
+            for s in await services_repo.list_services(
+                gids=[group.gid for group in user_groups],
+                execute_access=True,
+                write_access=True,
+                combine_access_with_and=False,
+                product_name=x_simcore_products_name,
+            )
+        }
     # Non-detailed views from the services_repo database
     if not details:
         # only return a stripped down version
         # NOTE: here validation is not necessary since key,version were already validated
         # in terms of time, this takes the most
-        return [
-            ServiceGet.construct(
-                key=key,
-                version=version,
-                name="nodetails",
-                description="nodetails",
-                type=ServiceType.COMPUTATIONAL,
-                authors=[{"name": "nodetails", "email": "nodetails@nodetails.com"}],
-                contact="nodetails@nodetails.com",
-                inputs={},
-                outputs={},
-                deprecated=services_in_db[(key, version)].deprecated,
-            )
-            for key, version in services_in_db
-        ]
+        with _tracer.start_as_current_span("return-non-detailed-view") as span:
+            span.set_attribute("osparc-gids", f"{[group.gid for group in user_groups]}")
+            span.set_attribute("osparc-productname", f"{x_simcore_products_name}")
```

Comment:
Q: I guess depending how we name these attributes, they will be followed more or less well in the tracing interface in the end right? would it not make sense at this point to have some central registry so that we do not create all kind of attribute names with 1 '-', multiple, wrong wordings, etc etc?

Reply:
Yes I fully agree. I had a similar thought but felt I have too little insight and experience with opentelemetry to standardize something at this point, but I would very much welcome a common attribute naming scheme